### PR TITLE
Implement analytics and projection view

### PR DESCRIPTION
**Pull request by andysmith26**
## Summary
- implement real scenario satisfaction metrics and cover edge cases with new analytics unit tests
- add dedicated student projection view route and link it from the program page
- polish analytics display and roster import instructions for teachers

## Testing
- pnpm test:unit *(fails: Playwright browsers not installed in environment)*

------
[Codex Task](https://chatgpt.com/codex/tasks/task_e_692cd0aee2bc8328b965e47098fefb7b)

### CODE REVIEW

File: src/lib/domain/analytics.ts [hunk 1]
```diff
@@ -21,20 +21,94 @@ export interface ScenarioSatisfaction {
...
+                                rank = i + 1;
+                                break;
+                        }
+                }
+
+                if (rank === 1) {
+                        topChoiceCount++;
+                }
+                if (rank <= 2) {
+                        top2Count++;
```
[the remark this answers is not in the document]
Comment (andysmith26) [reply src/lib/domain/analytics.ts]: ok for now
